tstest/integration/integration_test.go

```diff
@@ -224,6 +224,47 @@ func TestNodeAddressIPFields(t *testing.T) {
 	d1.MustCleanShutdown(t)
 }
 
+func TestAddPingRequest(t *testing.T) {
+	t.Parallel()
+	bins := BuildTestBinaries(t)
+
+	env := newTestEnv(t, bins)
+	defer env.Close()
+
+	n1 := newTestNode(t, env)
+	d1 := n1.StartDaemon(t)
+	defer d1.Kill()
+
+	n1.AwaitListening(t)
+	n1.MustUp()
+	n1.AwaitRunning(t)
+
+	gotPing := make(chan bool, 1)
+	waitPing := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) { gotPing <- true }))
```

Comment:
Sorry, I'd do this over three lines as I'd originally written on Slack:

https://tailscale.slack.com/archives/D0222E6J7DX/p1622746241009700

with `gotPing <- true` on its own line so it doesn't get lost in the noise of the one line.